src/main/java/com/github/pereiratostain/generator/MermaidSchemaGenerator.java

```diff
@@ -33,10 +42,90 @@ private void generateHeader(Writer writer) throws IOException {
   }
 
   private void generateEntity(Writer writer, Entity entity) throws IOException {
+    var associations = new ArrayList<String>();
+    var stereotype = "";
+    String fields;
+
+    if(entity.stereotype() == Entity.Stereotype.ENUM) {
+      fields = computeFieldsEnum(entity);
+      stereotype = "\t<<enumeration>>\n";
+    } else {
+      fields = computeFieldsClass(entity, associations);
+    }
+
     writer.append("    class ")
             .append(entity.name())
-            .append(" {")
-            .append("\n    }\n\n");
+            .append(" {\n")
+            .append(stereotype)
+            .append(fields)
+            .append("\n    }\n")
+            .append(generateAssociations(entity, associations))
+            .append("\n");
+  }
+
+  private String computeFieldsEnum(Entity entity) {
+    var fields = new ArrayList<Field>();
+
+    for(var field: entity.fields()) {
+      if(!field.name().equals("$VALUES")) {
+        fields.add(field);
+      }
+    }
+    return generateRecordFields(fields);
+  }
+
+  private String computeFieldsClass(Entity entity, ArrayList<String> associations) {
+    var fields = new ArrayList<Field>();
+    Pattern pattern = Pattern.compile("<.*>");
+
+    for(var field: entity.fields()) {
+      var fieldType = field.type();
+      fieldType = fieldType.replace(";", "");
```

Comment:
The signature can be parsed by the package signature of ASM instead of rewriting a buggy parser